pkg/apisix/cluster.go

```diff
@@ -301,6 +314,74 @@ func (c *cluster) HasSynced(ctx context.Context) error {
 	}
 }
 
+// syncSchema syncs schema from APISIX regularly according to the interval.
+func (c *cluster) syncSchema(ctx context.Context, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		if err := c.syncSchemaOnce(); err != nil {
+			log.Warnf("failed to sync schema: %s", err)
+		}
+
+		select {
+		case <-ticker.C:
+			continue
+		case <-ctx.Done():
+			break
+		}
+	}
+}
+
+// syncSchemaOnce syncs schema from APISIX once.
+// It firstly deletes all the schema in the cache,
+// then queries and inserts to the cache.
+func (c *cluster) syncSchemaOnce() error {
```

Comment:
context should be propagated here.

Reply:
then how to use it?

Reply:
`context.TODO()` are used in some places, you can replace them with the passed-in context.